wasm.md

```diff
@@ -863,27 +854,28 @@ Here, we allow for an "abstract" function declaration using syntax `func_::___`,
 
 ```k
     syntax Defn     ::= FuncDefn
-    syntax FuncDefn ::= "(" "func"            FuncExports TypeUse LocalDecls Instrs ")"
-                      | "(" "func" Identifier FuncExports TypeUse LocalDecls Instrs ")"
+    syntax FuncDefn ::= "(" "func" OptionalId FuncExports TypeUse LocalDecls Instrs ")"
  // -----------------------------------------------------------------------------------
-    rule <k> ( func FEXPO:FuncExports TUSE:TypeUse LDECLS:LocalDecls INSTRS:Instrs )
+    rule <k> ( func ID:OptionalId FEXPO:FuncExports TUSE:TypeUse LDECLS:LocalDecls INSTRS:Instrs )
           => ( func #freshId(NEXTID) FEXPO TUSE LDECLS INSTRS ) ...
          </k>
          <nextFreshId> NEXTID => NEXTID +Int 1 </nextFreshId>
+      requires notBool isIdentifier(ID)
 
     rule <k> ( func FNAME:Identifier ( export ENAME ) FEXPO:FuncExports TUSE:TypeUse LDECLS:LocalDecls INSTRS:Instrs )
           => ( export ENAME ( func FNAME ) )
           ~> ( func FNAME FEXPO TUSE LDECLS INSTRS )
           ...
          </k>
 
-    rule <k> ( func FNAME:Identifier .FuncExports TUSE:TypeUse LDECLS:LocalDecls INSTRS:Instrs ) => #checkTypeUse ( TUSE ) ... </k>
+    rule <k> ( func FNAME:Identifier .FuncExports TUSE:TypeUse LDECLS:LocalDecls INSTRS:Instrs )
+          => saveId("func", FNAME, NEXTIDX)
+          ~> #checkTypeUse ( TUSE ) ... </k>
```

Comment:
`...` on it's own line, `</k>` on its own line.

Reply:
removed `saveId`, changed back to one line